flair/models/biomedical_entity_linking.py

```diff
@@ -0,0 +1,988 @@
+    """
+
+    def __init__(self, dictionary_path: Union[Path, str]) -> None:
+        """
+        :param dictionary_path str: The path of the dictionary
+        """
+        log.info("Loading Dictionary from {}".format(dictionary_path))
+        self.data = self.load_data(dictionary_path)
+
+    def load_data(self, dictionary_path) -> np.ndarray:
+        data = []
+        with open(dictionary_path, mode="r", encoding="utf-8") as f:
+            lines = f.readlines()
+            for line in tqdm(lines, desc="Loading dictionary"):
+                line = line.strip()
+                if line == "":
+                    continue
+                cui, name = line.split("||")
+                name = name.lower()
+                data.append((name, cui))
+
+        data = np.array(data)
+        return data
+
+
+class BiEncoderEntityLiker:
+    """
+    A class to load a model and use it to encode a dictionary and entities
+    """
+
+    def __init__(
+        self, use_sparse_embeds: bool, max_length: int, index_use_cuda: bool
+    ) -> None:
+        self.use_sparse_embeds = use_sparse_embeds
+        self.max_length = max_length
+
+        self.tokenizer = None
+        self.encoder = None
+
+        self.sparse_encoder = None
+        self.sparse_weight = None
+
+        self.index_use_cuda = index_use_cuda and flair.device.type == "cuda"
+
+        self.dense_dictionary_index = None
+        self.dict_sparse_embeds = None
+
+        self.dictionary = None
+
+    def load_model(
+        self,
+        model_name_or_path: Union[str, Path],
+        dictionary_name_or_path: str,
+        mean_centering: bool,
+        batch_size: int = 1024,
+    ):
+        """
+        Load the model and embed the dictionary
+        :param model_name_or_path: The path of the model
+        :param dictionary_name_or_path: The path of the dictionary
+        :param mean_centering: Whether to mean center the dictionary embeddings
+        :param batch_size: The batch size for embedding the dictionary
+        """
+        self.load_dense_encoder(model_name_or_path)
+
+        if self.use_sparse_embeds:
+            self.load_sparse_encoder(model_name_or_path)
+            self.load_sparse_weight(model_name_or_path)
+
+        self.embed_dictionary(
+            model_name_or_path=model_name_or_path,
+            dictionary_name_or_path=dictionary_name_or_path,
+            mean_centering=mean_centering,
+            batch_size=batch_size,
+        )
+
+        return self
+
+    def load_dense_encoder(
+        self, model_name_or_path: Union[str, Path]
+    ) -> Tuple[PreTrainedModel, PreTrainedTokenizer]:
+
+        self.encoder = TransformerDocumentEmbeddings(
+            model_name_or_path, is_token_embedding=False
+        )
+
+        return self.encoder
+
+    def load_sparse_encoder(
+        self, model_name_or_path: Union[str, Path]
+    ) -> SparseWordVectorizer:
+        sparse_encoder_path = os.path.join(model_name_or_path, "sparse_encoder.pk")
```

Comment:
"sparse_encoder.pkl"?
I vaguely remember that we talked about some "security concerns" when loading pickle objects...